hnn_core/pyramidal.py

```diff
@@ -219,6 +219,43 @@ def _synapse_create(self, p_syn):
                 self.dends['apical_tuft'](0.5), **p_syn['gabaa'])
 
 
+    def _connect_at_loc(self, loc, receptor, gid_src, nc_dict, nc_list):
+        """Get list of proximal/distal postsynaptic junctions.
+
+        Parameters
+        ----------
+        loc : str
+            Either 'proximal' or 'distal'
+        receptor : str
+            'nmda', 'ampa' etc.
+        gid_src : int
+            The Cell ID of the source.
+        nc_dict : dict.
+            The connection parameters. Usually contains the keys
+            pos_src, A_weight, A_delay, lamtha.
+        nc_list : list of NetCon
+            Different intput types are appended to different lists
+            of NetCon objects. The created network connections will
+            be appended to this list.
+        """
+        if loc == 'proximal':
+            dends = ['apicaloblique', 'basal2', 'basal3']
+        elif loc == 'distal':
+            dends = ['apicaltuft']
+        else:
+            raise ValueError('loc must be one of proximal or distal')
+
+        postsyns = list()
+        for dend in dends:
+            postsyns.append(getattr(self, f'{dend}_{receptor}'))
+
+        for postsyn in postsyns:
+            nc = self.parconnect_from_src(gid_src, nc_dict, postsyn)
+            nc_list.append(nc)
+
+        return postsyns
```

Comment:
So the main point here is to simplify the `parreceive()` and `parreceive_ext()` methods correct? If so, I definitely like how this PR focuses on consolidating redundant code.

Reply:
yes, but also to make the code a bit more readable. Not just concise but more readable. We'll probably need to do a couple more iterations / PRs to get it absolutely on point.